src/test/java/org/fcrepo/client/ConnectionManagementIT.java

```diff
@@ -0,0 +1,371 @@
+            assertNotNull("Expected a non-null InputStream.", response.getBody());
+            try {
+                IOUtils.copy(response.getBody(), NullOutputStream.NULL_OUTPUT_STREAM);
+            } catch (IOException e) {
+                // ignore
+            }
+        };
+
+    }
+
+    /**
+     * The Fedora Repository client.
+     */
+    private FcrepoClient client;
+
+    /**
+     * The Apache HttpClient under test.
+     */
+    private CloseableHttpClient underTest;
+
+    /**
+     * The {@link org.apache.http.conn.HttpClientConnectionManager} implementation that the {@link #underTest
+     * HttpClient} is configured to used.
+     */
+    @Spy
+    private PoolingHttpClientConnectionManager connectionManager = new PoolingHttpClientConnectionManager();
+
+    @Before
+    public void setUp() throws Exception {
+
+        // Required because we have a test that doesn't close connections, so we have to insure that the
+        // connection manager doesn't block during that test.
+        connectionManager.setDefaultMaxPerRoute(HttpMethods.values().length);
+
+        // Set up the expectations on the Mock http server
+        new MockHttpExpectations().initializeExpectations(this.mockServerClient, this.mockServerRule.getPort());
+
+        // Uris that we connect to, and answered by the Mock http server
+        uris = new MockHttpExpectations.SupportedUris();
+
+        // A FcrepoClient configured to throw exceptions when an error is encountered.
+        client = new FcrepoClient(null, null, host + ":" + port, true);
+
+        // We're testing the behavior of a default HttpClient with a pooling connection manager.
+        underTest = HttpClientBuilder.create().setConnectionManager(connectionManager).build();
+
+        // Put our testable HttpClient instance on the FcrepoClient
+        setField(client, "httpclient", underTest);
+
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        underTest.close();
+    }
+
+    /**
+     * Demonstrates that HTTP connections are released when the FcrepoClient throws an exception.  Each method of the
+     * FcrepoClient (get, put, post, etc.) is tested.
+     *
+     * @throws Exception if something exceptional happens
+     */
+    @Test
+    public void connectionReleasedOnException() throws Exception {
+        // Hard-coded 6 b/c HttpMethods lists Options, which isn't supported by the FcrepoClient.
+        final int expectedCount = 6;
+        int actualCount = 0;
+
+        for (HttpMethods method : HttpMethods.values()) {
+            if (method == HttpMethods.OPTIONS) {
+                continue;
+            }
+            connect(client, uris.uri500, method, null);
+            actualCount++;
+        }
+
+        assertEquals("Expected to make " + expectedCount + " connections; made " + actualCount,
+                expectedCount, actualCount);
+
+        Verifications.connectionRequestedAndClosed(actualCount).accept(connectionManager);
```

Comment:
The reason I didn't is because `Verifications.` is essentially a comment disguised as a class name, but happy to change.